db/sql.php

```diff
@@ -122,12 +120,7 @@ function type($val) {
 	function value($type,$val) {
 		switch ($type) {
 			case self::PARAM_FLOAT:
-				if (is_string($val)) {
-					$val=str_replace(',','.',preg_replace('/([.,])(?!\d+$)/','',$val));
-					if (strlen($val)-(int)(bool)strpos($val,'.') <= $this->precision)
-						$val=(float) $val;
-				}
-				return $val;
+				return rtrim(sprintf('%F',$val),'0');
```

Comment:
Actually it is a valid format and `1.` makes it clear that it's a float (and not an integer).  But I guess we don't need it. So I've just switched `sprintf` with a simple `str_replace`.

Also I've restricted the processing to `float` only, so that values coming from the db remain untouched.